src/expr.rs

```diff
@@ -854,19 +829,58 @@ fn rewrite_call(context: &RewriteContext,
                 -> Option<String> {
     debug!("rewrite_call, width: {}, offset: {}", width, offset);
 
-    // FIXME using byte lens instead of char lens (and probably all over the place too)
     // 2 is for parens
-    let max_callee_width = try_opt!(width.checked_sub(2));
-    let callee_str = try_opt!(callee.rewrite(context, max_callee_width, offset));
-    debug!("rewrite_call, callee_str: `{}`", callee_str);
-
-    if args.is_empty() {
-        return Some(format!("{}()", callee_str));
+    let mut hi = try_opt!(width.checked_sub(2)) * 2;
+    let mut lo = 0;
+    let mut tries = 0;
+
+    // Binary search for the best split between callee and arguments.
+    loop {
+        let middle = (lo + hi) / 2;
+
+        match rewrite_call_inner(context, callee, middle, args, span, width, offset) {
+            _ if tries > 10 => return None,
+            Ok(result) => return Some(result),
+            Err(Ordering::Less) => {
+                lo = middle;
```

Comment:
You're absolutely right! I guess this proves @nrc's point about binary search being prone to off-by-one errors ;-)